src/wazuh_modules/agent_upgrade/manager/wm_agent_upgrade_validate.c

```diff
@@ -415,3 +415,27 @@ bool wm_agent_upgrade_validate_task_status_message(const cJSON *response, char *
     }
     return false;
 }
+
+bool wm_agent_upgrade_validate_task_ids_message(const cJSON *input_json, int *agent_id, int *task_id, char** data) {
+    if (input_json) {
+        cJSON *agent_json = cJSON_GetObjectItem(input_json, task_manager_json_keys[WM_TASK_AGENT_ID]);
+        cJSON *data_json = cJSON_GetObjectItem(input_json, task_manager_json_keys[WM_TASK_ERROR_DATA]);
+        cJSON *task_json = cJSON_GetObjectItem(input_json, task_manager_json_keys[WM_TASK_TASK_ID]);
+
+        if (agent_json && (agent_json->type == cJSON_Number)) {
+            *agent_id = agent_json->valueint;
+        } else {
+            return false;
+        }
+
+        if (data_json && (data_json->type == cJSON_String)) {
+            os_strdup(data_json->valuestring, *data);
+        }
+
+        if (task_json && (task_json->type == cJSON_Number)) {
+            *task_id = task_json->valueint;
+        }
+        return true;
+    }
```

Comment:
There seems to be no reason for this, if this method return false it will trigger the error section of the callback method which will end up executing that same line

Reply:
Ok, so you can remove the error in line 414?

Reply:
Done